### PR TITLE
Implement mget

### PR DESCRIPTION
**Pull request by shailesh33**
This patch adds support for mget in the client.
It doesnt do the real "scatter gather" of mget yet.

### CODE REVIEW

**Comment by ipapapa** [comment]
* Is there a specific reason on why compression was not added? What happens if somebody uses MSET with the compression FP enabled, and then tries to do a simple GET?
* Can you shrink the number of commits?

**Comment by shailesh33** [comment]
@ipapapa the branch was merged /updated from master so I cannot merge the commits easily.

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java [hunk 1]
```diff
@@ -2430,8 +2515,31 @@ public Long pexpire(String key, long milliseconds) {
...
+        if (CompressionStrategy.NONE == connPool.getConfiguration().getCompressionStrategy()) {
+
+            return connPool.executeWithFailover(new MultiKeyOperation<List<String>>(new ArrayList<String>(Arrays.asList(keys)), OpName.MGET) {
+                @Override
+                public List<String> execute(Jedis client, ConnectionContext state) {
+                    return client.mget(keys);
+                }
+            });
+        } else {
+            return connPool.executeWithFailover(new CompressionValueMultiKeyOperation<List<String>>(new ArrayList<String>(Arrays.asList(keys)), OpName.MGET) {
```
Comment (jcacciatore) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java]: Same comment as above

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java [hunk 2]
```diff
@@ -2430,8 +2515,31 @@ public Long pexpire(String key, long milliseconds) {
...
 
     @Override
-    public List<String> mget(String... keys) {
-        throw new UnsupportedOperationException("not yet implemented");
+    public List<String> mget(String... keys) { return d_mget(keys).getResult(); }
+
+    public OperationResult<List<String>> d_mget(final String... keys) {
+        if (CompressionStrategy.NONE == connPool.getConfiguration().getCompressionStrategy()) {
+
+            return connPool.executeWithFailover(new MultiKeyOperation<List<String>>(new ArrayList<String>(Arrays.asList(keys)), OpName.MGET) {
```
Comment (shailesh33) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java]: took care of it.

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java (hunk 1 above)
Comment (shailesh33) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java]: done

**Comment by shailesh33** [comment]
Thank you @jcacciatore for your review. I appreciate you taking the time.
The mget receives a null for a (nil) for a key that is missing. So that works.
If one of the node is down, an exception is thrown by the client.

**Comment by jcacciatore** [comment]
Cool, looks good @shailesh33. The only other suggestion I have is to document the mget behavior, specifically for failure scenarios, in a javadoc header. Other than that LGTM

**Comment by jcacciatore** [comment]
Didn't mean to close the PR...

**Pull request merged**